### ISSUE TITLE
Robot Object for `compas_viewer`

### ISSUE THREAD

**Issue by ZacZhangzhuo** [compas-dev/compas_robots#2]
`compas_viewer` is ready to connect `compas_robots` for visualizing the robot.

Since I saw the main updates are in the `refactor_2.0` branch and not merged into `main`, I thought it would be great to create an issue to discuss first before I start.

The template would be like;

```python
from compas_viewer.scene.meshobject import MeshObject

class RobotObject(??MeshObject??):
    """Object for displaying COMPAS RobotModel.    """

    def __init__(self, robot, **kwargs):

    def _read_points_data(self) -> Optional[DataType]:
        """Read points data from the object."""
        pass


    def _read_lines_data(self) -> Optional[DataType]:
        """Read lines data from the object."""
        pass

    def _read_frontfaces_data(self) -> Optional[DataType]:
        """Read frontfaces data from the object."""
        pass

    def _read_backfaces_data(self) -> Optional[DataType]:
        """Read backfaces data from the object."""
        pass

```

Would also be great if you could point out where is the port that I should connect :)

**Comment by gonzalocasas** [comment]
Good idea, the robot model objects to support `Scene` are relatively simple to create. Most of the heavy lifting is done in a base class that works independent of CAD (the `compas_robots.scene.BaseRobotModelObject`), it handles all the things transforming the links of the robot to match the configuration, and so on, so, to add support for a new viewer or CAD, there are only a couple of methods needed that are defined by `AbstractRobotModelObject`:

```
class AbstractRobotModelObject(object):
    def transform(self, geometry, transformation):
        raise NotImplementedError

    def create_geometry(self, geometry, name=None, color=None):
        raise NotImplementedError
```

so, we just need to have a method to create native geometry in the target visualization tool, and one to natively transform geometry.

The `RobotModelObject` for `ghpython` is a good example of this because it is simple to understand.

In the end, due to the methods inherited from the base class, the robot model scene objects end up having the following relevant methods (related to your template): `draw_visual`, `draw_collision`, `draw_attached_meshes`.

**Comment by yck011522** [comment]
@ZacZhangzhuo Is this issue already been resolved? It seems the PR is merged

**Comment by jf---** [comment]
@yck011522 you're right lets see if @gonzalocasas or @tomvanmele can close it?
